### ISSUE TITLE
Remove create-tag branch condition on default branch

### ISSUE THREAD

**Issue by sqin2019** [abcxyz/pkg#215]
To prevent creating tags on unreviewed tag, we now only allow tags on repo default branch (main) since

1. we rely on deployment to enforce workflow review
2. but currently it does not prevent self-review. 

Once prevent self-review is done, see tracking issue https://github.com/github/roadmap/issues/825, we can remove the condition.

**Comment by sethvargo** [comment]
Hi @sqin2019 - do you have more information on what we need to do here? I'm not following how the "prevent self-review" option is going to prevent someone from creating and pushing a tag locally with the git CLI.

**Comment by sqin2019** [comment]
Hi @sethvargo - the action item here is to remove https://github.com/abcxyz/pkg/blob/main/.github/workflows/create-tag.yml#L173, currently it only allows people to create tags on main so that even though I can approve my own create tag deployment, I will be creating a tag on protected branch. I believe prevent self-review is already done by GitHub, this is not necessary anymore, we will rely on deployment reviewer to check if the tag is created on unprotected branches for valid reasons. 

With protected tag org ruleset to disallow creating any tag unless bypass, are we still be able to create and push tag locally with git CLI?

**Comment by sethvargo** [comment]
I think we may be talking about different things. It's currently possible to `git tag && git push origin --tags` from most of our repositories.

**Comment by sethvargo** [comment]
Sorry, it's currently _impossible_ to `git tag && git push origin --tags` (I was typing too quickly). I think this issue is suggesting that we _remove_ that constraint, which concerns me. Am I misunderstanding?

**Comment by sqin2019** [comment]
> Sorry, it's currently _impossible_ to `git tag && git push origin --tags` (I was typing too quickly). I think this issue is suggesting that we _remove_ that constraint, which concerns me. Am I misunderstanding?

it is this PR https://github.com/abcxyz/pkg/pull/241.

**Comment by sethvargo** [comment]
So we're just removing it from the workflow, but not the tag ruleset?

**Comment by sqin2019** [comment]
> So we're just removing it from the workflow, but not the tag ruleset?

Yes, just the workflow.
